src/main/java/org/wildfly/security/auth/server/SecurityDomain.java

```diff
@@ -309,16 +309,15 @@ PermissionCollection mapPermissions(SecurityIdentity securityIdentity) {
         return this.permissionMapper.mapPermissions(principal, roles);
     }
 
-    String getDefaultRealmName() {
-        return defaultRealmName;
-    }
-
     NameRewriter getPreRealmRewriter() {
         return preRealmRewriter;
     }
 
     RealmMapper getRealmMapper() {
-        return realmMapper;
+        return (String userName) -> {
```

Comment:
This should be cached in a field then.  Better still though would be a simple "mapRealmName" method which would do the mapping and the if check without using a lambda.